server/util/log/log.go

```diff
@@ -111,24 +117,46 @@ func StructuredLogger() zerolog.Logger {
 	return zerolog.New(os.Stdout).With().Timestamp().Logger()
 }
 
-func Configure(level string, enableFileName, enableStructured bool) error {
+type gcpLoggingCallerHook struct{}
+
+func (h gcpLoggingCallerHook) Run(e *zerolog.Event, level zerolog.Level, msg string) {
+	_, file, line, ok := runtime.Caller(callerSkipFrameCount)
+	if !ok {
+		return
+	}
+	sourceLocation := zerolog.Dict().Str("file", filepath.Base(file)).Str("line", strconv.Itoa(line))
+	e.Dict("logging.googleapis.com/sourceLocation", sourceLocation)
+}
+
+type Opts struct {
+	Level                  string
+	EnableShortFileName    bool
+	EnableGCPLoggingFormat bool
+	EnableStructured       bool
+}
+
+func Configure(opts Opts) error {
 	var logger zerolog.Logger
-	if enableStructured {
+	if opts.EnableStructured {
 		logger = StructuredLogger()
 	} else {
-		logger = LocalLogger(level)
+		logger = LocalLogger(opts.Level)
```

Comment:
I don't think LocalLogger needs level anymore -- can you remove the need for this arg?

Reply:
Done.